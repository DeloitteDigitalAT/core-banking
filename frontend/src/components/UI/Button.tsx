import React from "react";

interface ButtonType {
  text: string;
  handler: () => void;
  active: boolean;
}

const Button: React.FC<ButtonType> = properties => {
  const { text, handler, active } = properties;
  return (
    <button
      type="button"
      className={`${
        active ? "bg-green-500 border-green-500 hover:bg-green-700" : "bg-green-200 "
      } border   text-white rounded-md px-6 py-2 my-4 transition duration-500 ease select-none focus:outline-none focus:shadow-outline`}
      onClick={handler}
    >
      {text}
    </button>
  );
};

export default Button;
