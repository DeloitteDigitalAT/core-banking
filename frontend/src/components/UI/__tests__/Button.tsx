import React from "react";
import renderer from "react-test-renderer";
import Button from "../Button";

describe("UI/Button component", () => {
  test("snapshot renders", () => {
    const tree = renderer.create(<Button text="" handler={() => {}} active />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
