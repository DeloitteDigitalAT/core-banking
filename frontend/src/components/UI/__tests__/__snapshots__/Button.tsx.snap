// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Button component snapshot renders 1`] = `
<button
  className="bg-gradient-to-b from-green-400 via-green-500 to-green-600 border-green-500 hover:from-green-600 hover:via-green-700 hover:to-green-800 text-sm border text-white rounded-md px-6 py-2 my-4 transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
  onClick={[Function]}
  type="button"
>
  
</button>
`;
